ansible/roles/common/tasks/main.yml

```diff
@@ -10,6 +10,39 @@
     state: latest
   when: update_packages|bool
 
+#- name: Reboot all VMs after updating to the latest release
+#  reboot:
+#    connect_timeout: 300
+#    msg: "Ansible updates installing. Rebooting now."
+#    pre_reboot_delay: 60
+#    post_reboot_delay: 10
+#  when:
+#  - update_packages|bool
+#  - ansible_version.full is version_compare("2.7", ">=")
+
+- name: Check if right module is used
+  fail:
+    msg: Please use the reboot module in Ansible > 2.7
+  when:
+  - ansible_version.full is version_compare("2.7", ">=")
+
+- name: Reboot all VMs after updating to the latest release
+  shell: "sleep 2 && shutdown -r now "
+  async: 1
+  poll: 0
+  ignore_errors: true
+  when:
+  - update_packages|bool
+  - ansible_version.full is version_compare("2.7", "<")
+
+- name: Wait for VMs to come back up
+  wait_for_connection:
+    delay: 30
+    timeout: 300
+  when:
+  - update_packages|bool
+  - ansible_version.full is version_compare("2.7", "<")
```

Comment:
```#- name: Reboot all VMs after updating to the latest release
#  reboot:
#    connect_timeout: 300
#    msg: "Ansible updates installing. Rebooting now."
#    pre_reboot_delay: 60
#    post_reboot_delay: 10
#  when:
#  - update_packages|bool
#  - ansible_version.full is version_compare("2.7", ">=")

- name: Check if right module is used
  fail:
    msg: Please use the reboot module in Ansible > 2.7
  when:
  - ansible_version.full is version_compare("2.7", ">=")
```

You commented the ansible-2.7 region here. If i understand correctly, if a user has ansible 2.7 and is using the `common` role, his playbook will fail. Then he/she will have to uncomment the 2.7 region of the code, and comment the 2.6 region of the code, right ?

I would prefer if the common module could work for both 2.6 and 2.7, without manual steps for the user.

I think this is achievable by using dynamic includes so ansible 2.6 doesn't throw a syntax error if it sees `reboot` module somewhere. For example, have you tried:

```
- name: Reboot all VMs after updating to the latest release (ansible 2.7)
  include_tasks: reboot_27.yml
  when: ansible_version.full is version_compare("2.7", ">=")

- name: Reboot all VMs after updating to the latest release (ansible 2.6)
  include_tasks: reboot_26.yml
  when: ansible_version.full is version_compare("2.7", "<")